app/src/main/java/org/fossasia/pslab/communication/sensors/ADS1115.java

```diff
@@ -0,0 +1,243 @@
+    private int REG_CONFIG_DR_860SPS = 7<<5;
+
+    private int REG_CONFIG_CMODE_MASK = 0x0010;
+    private int REG_CONFIG_CMODE_TRAD = 0x0000;
+    private int REG_CONFIG_CMODE_WINDOW = 0x0010;
+
+    private int REG_CONFIG_CPOL_MASK = 0x0008;
+    private int REG_CONFIG_CPOL_ACTVLOW = 0x0000;
+    private int REG_CONFIG_CPOL_ACTVHI = 0x0008;
+
+    private int REG_CONFIG_CLAT_MASK = 0x0004;
+    private int REG_CONFIG_CLAT_NONLAT = 0x0000;
+    private int REG_CONFIG_CLAT_LATCH = 0x0004;
+
+    private int REG_CONFIG_CQUE_MASK = 0x0003;
+    private int REG_CONFIG_CQUE_1CONV = 0x0000;
+    private int REG_CONFIG_CQUE_2CONV = 0x0001;
+    private int REG_CONFIG_CQUE_4CONV = 0x0002;
+    private int REG_CONFIG_CQUE_NONE = 0x0003;
+
+    private HashMap <String,Integer> gains = new HashMap <String,Integer>();
+    private HashMap <String,Double> gainScaling = new HashMap <String,Double>();
+    private HashMap <String,String> typeSelection = new HashMap <String,String>();
+    private HashMap <Integer,Integer> sdrSelection = new HashMap <Integer,Integer>();
+
+    private String channel;
+    private String gain;
+    private int rate;
+
+    public int NUMPLOTS = 1;
+    public String[] PLOTNAMES = {"mV"};
+
+    public ADS1115 (I2C i2c) throws IOException, InterruptedException{
+        this.i2c = i2c;
+        channel = "UNI_0";
+        gain = "GAIN_ONE";
+        rate = 128;
+
+        setGain("GAIN_ONE");
+        setChannel("UNI_0");
+        setDataRate(128);
+
+        int conversionDelay = 8;
+        String name = "ADS1115 16-bit ADC";
+
+        gains.put("GAIN_TWOTHIRDS",REG_CONFIG_PGA_6_144V);
+        gains.put("GAIN_ONE",REG_CONFIG_PGA_4_096V);
+        gains.put("GAIN_TWO",REG_CONFIG_PGA_2_048V);
+        gains.put("GAIN_FOUR",REG_CONFIG_PGA_1_024V);
+        gains.put("GAIN_EIGHT",REG_CONFIG_PGA_0_512V);
+        gains.put("GAIN_SIXTEEN",REG_CONFIG_PGA_0_256V);
+
+        gainScaling.put("GAIN_TWOTHIRDS",0.1875);
+        gainScaling.put("GAIN_ONE",0.125);
+        gainScaling.put("GAIN_TWO",0.0625);
+        gainScaling.put("GAIN_FOUR",0.03125);
+        gainScaling.put("GAIN_EIGHT",0.015625);
+        gainScaling.put("GAIN_SIXTEEN",0.0078125);
+
+        typeSelection.put("UNI_0","0");
+        typeSelection.put("UNI_1","1");
+        typeSelection.put("UNI_2","2");
+        typeSelection.put("UNI_3","3");
+        typeSelection.put("DIFF_01","01");
+        typeSelection.put("DIFF_23","23");
+
+        sdrSelection.put(8,REG_CONFIG_DR_8SPS);
+        sdrSelection.put(16,REG_CONFIG_DR_16SPS);
+        sdrSelection.put(32,REG_CONFIG_DR_32SPS);
+        sdrSelection.put(64,REG_CONFIG_DR_64SPS);
+        sdrSelection.put(128,REG_CONFIG_DR_128SPS);
+        sdrSelection.put(250,REG_CONFIG_DR_250SPS);
+        sdrSelection.put(475,REG_CONFIG_DR_475SPS);
+        sdrSelection.put(860,REG_CONFIG_DR_860SPS);
+
+    }
+
+    public short __readInt__(int addr) throws IOException, InterruptedException {
```

Comment:
one more thing readUInt is returning a "short" which readInt is casting as a "short" ( doesn't make any difference ) and returning again.
change readUInt to readInt ( get rid of readInt function ) and return normal "int" as lower bits of short are still there in int and we have been using "int" for porting.
This function is similar to readRegister in which you read 2 bytes and returned "int" from them.  

Thanks